lib/include/jxl/encode.h

```diff
@@ -264,8 +264,123 @@ JXL_EXPORT JxlEncoderStatus JxlEncoderAddImageFrame(
+ *   split across multiple boxes. The codestream contains the JPEG XL image
+ *   itself, including the basic info such as image dimensions, ICC color
+ *   profile, and all the pixel data of all the image frames.
+ * - "jxlp": a codestream box in case it is split across multiple boxes. The
+ *   encoder will automatically do this if necessary. The contents are the same
+ *   as in case of a jxlc box, when concatenated.
+ * - "Exif": a box with EXIF metadata, can be added by libjxl users, or is
+ *   automatically added when needed for JPEG reconstruction. The contents of
+ *   this box must be prepended by a 4-byte tiff header offset, which may
+ *   be 4 zero bytes.
+ * - "XML ": a box with XMP or IPTC metadata, can be added by libjxl users, or
+ *   is automatically added when needed for JPEG reconstruction
+ * - "jumb": a JUMBF superbox, which can contain boxes with different types of
+ *   metadata inside. This box type can be added by the encoder transparently,
+ *   and other libraries to create and handle JUMBF content exist.
+ * - "brob": a Brotli-compressed box, which otherwise represents an existing
+ *   type of box such as Exif or XML. The encoder creates these when enabled and
+ *   users of libjxl don't need to create them directly. Some box types are not
+ *   allowed to be compressed: any of the signature, jxl* and jbrd boxes.
+ * - "jxli": frame index box, can list the keyframes in case of a JXL animation,
+ *   allowing the decoder to jump to individual frames more efficiently. This
+ *   box type is specified, but not currently supported by the encoder or
+ *   decoder.
+ * - "jbrd": JPEG reconstruction box, contains the information required to
+ *   byte-for-byte losslessly recontruct a JPEG-1 image. The JPEG coefficients
+ *   (pixel content) themselves are encoded in the JXL codestream (jxlc or jxlp)
+ *   itself. Exif and XMP metadata will be encoded in Exif and XMP boxes. The
+ *   jbrd box itself contains information such as the app markers of the JPEG-1
+ *   file and everything else required to fit the information together into the
+ *   exact original JPEG file. This box is added automatically by the encoder
+ *   when needed, and only when JPEG reconstruction is used.
+ * - other: other application-specific boxes can be added. Their typename should
+ *   not begin with "jxl" or "JXL" or conflict with other existing typenames.
+ *
+ * Most boxes are automatically added by the encoder and should not be added
+ * with JxlEncoderAddBox. Boxes that one may wish to add with JxlEncoderAddBox
+ * are: Exif and XML (but not when using JPEG reconstruction since if the
+ * JPEG has those, these boxes are already added automatically), jumb, and
+ * application-specific boxes.
+ *
+ * Adding metadata boxes increases the filesize. When adding Exif metadata, the
+ * data must be in sync with what is encoded in the JPEG XL codestream,
+ * specifically the image orientation.
```

Comment:
Could add something like: `In case of conflicting metadata, the JPEG XL codestream takes precedence.`

Reply:
Done, but pointed out it's not recommended to do that, encoders should be encouraged to keep them in sync, it's more a decoder feature